### ISSUE TITLE
Explanation Warning message

### ISSUE THREAD

**Issue by LeaBro** [ICB-DCM/pyABC#574]
Hi,

as I was using the pyabc-package for parameter fitting, I had to play around with different bounds for prior parameters and tried different datasets. I am using lognormal priors and the multivariate transition kernel. At different occasions, I ran into this warning message:

"WARNING:pyabc.util:Unusually many (model, parameter) samples have prior density zero. The transition might be inappropriate."

Since this warning did not seem to obstruct the run itself, I could continue my analysis. Anyway, it might be useful to learn more about the reason behind it. Can anyone explain to me how this warning effects the process/results of parameter fitting and what the trigger is for it to show up? It has been difficult to find where the warning is raised within the source code.

Greets,
Lea

**Comment by yannikschaelte** [comment]
Hi, that warning would be raised here https://github.com/ICB-DCM/pyABC/blob/main/pyabc/inference_util.py#L191. What happens here is that parameters are sampled from the transition kernel, and then compared to the prior to make sure the prior is positive, as otherwise that particle would be assigned a zero weight later. This warning may happen e.g. if your prior is bounded, and the transition suggests many values outside the boundaries, or e.g. if you have discrete parameters.

**Comment by LeaBro** [comment]
Hi,
does this imply the transition kernel is wanting to sample values from outside the bounds that I have specified? I thought in that case it could help to use wider bounds, but it did not.
Do you know a general solution to this issue? Or should I continue to ignore it?

**Comment by yannikschaelte** [comment]
Hi, this would be one explanation. You could plot the posterior distributions you obtain vs the prior, to see whether these concentrate (for single parameters) on the edge.

I would advise to not ignore the issue, but to see what is happening, as it never occurred for me, and could indicate sth is wrong (it used to be an error, just recently became a warning).

**Comment by thejasvibr** [comment]
Hi, I'm writing with regards to the same warning. 
I have a simulation model that is very intensive to run, and so I'm hoping to do parameter inference on 'R', from a discrete set of parameter values in the 'R_val' np.array. Here I tried to formulate the indices of each R value as a ```scipy.randint``` distribution. 

```
priors_v2 = pyabc.Distribution(calldirection=pyabc.RV("uniform", 0, 2*np.pi*0.9),
                           R_ind=pyabc.RV("randint", 0, R_val.size))
```
However, for some reason the ABCSMC run seems to get stuck after the warning and doesn't show progress. 

Are there are any alternative formulations I can use to perform parameter inference on discrete variables? I did manage to 'fool' the ABCSMC by treating 'R' as a truly continuous variable, but the simulation chooses the nearest point on the grid. I'm not sure if this adds unnecessary computation and may provide misleading results?

I'm running a Windows 10, pyabc version 0.12.13
